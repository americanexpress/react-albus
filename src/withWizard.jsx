/*
 * Copyright (c) 2017 American Express Travel Related Services Company, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License
 * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing permissions and limitations under
 * the License.
 */

import React from 'react';
import PropTypes from 'prop-types';
import hoistStatics from 'hoist-non-react-statics';
import useWizard from './hooks/useWizard';

const withWizard = Component => {
  const WithWizard = props => {
    const wizard = useWizard();

    return <Component wizard={wizard} {...props} />;
  };

  WithWizard.contextTypes = {
    wizard: PropTypes.object,
  };

  WithWizard.displayName = `withWizard(${Component.displayName || Component.name})`;
  WithWizard.WrappedComponent = Component;

  return hoistStatics(WithWizard, Component);
};

export default withWizard;
