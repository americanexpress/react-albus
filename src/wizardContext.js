import React from "react";

const WizardContext = React.createContext({});

export default WizardContext;
