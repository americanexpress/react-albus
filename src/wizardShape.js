/*
 * Copyright (c) 2017 American Express Travel Related Services Company, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License
 * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing permissions and limitations under
 * the License.
 */

import PropTypes from "prop-types";
import historyShape from "./historyShape";

export default PropTypes.shape({
  go: PropTypes.func.isRequired,
  set: PropTypes.func.isRequired,
  history: historyShape,
  onNext: PropTypes.func.isRequired,
  onPrevious: PropTypes.func.isRequired,
  hasNext: PropTypes.bool.isRequired,
  hasPrevious: PropTypes.bool.isRequired,
  push: PropTypes.func.isRequired,
  replace: PropTypes.func.isRequired,
  step: PropTypes.shape({
    id: PropTypes.string.isRequired
  }).isRequired,
  steps: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired
    }).isRequired
  ).isRequired
});
