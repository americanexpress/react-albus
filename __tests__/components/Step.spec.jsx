/*
 * Copyright (c) 2017 American Express Travel Related Services Company, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License
 * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing permissions and limitations under
 * the License.
 */

import React from "react";
import { shallow } from "enzyme";
import { Step } from "../../src";

const mockWizard = { drinkMore: "butter beer" };

jest.mock("../../src/hooks/useWizard", () => jest.fn(() => mockWizard));

describe("Step", () => {
  it("should render children", () => {
    const rendered = shallow(
      <Step>
        <div />
      </Step>
    );

    expect(rendered).toMatchSnapshot();
  });

  it("should pass wizard to function as child", () => {
    shallow(
      <Step>
        {wizard => {
          expect(wizard).toEqual(mockWizard);
        }}
      </Step>
    );
  });

  it("should pass wizard to render prop", () => {
    shallow(
      <Step
        render={wizard => {
          expect(wizard).toEqual(mockWizard);
        }}
      />
    );
  });
});
