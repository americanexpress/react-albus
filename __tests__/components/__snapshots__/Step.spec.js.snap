// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step should render children 1`] = `<div />`;
